linkerd/meshtls/rustls/src/creds/verify.rs

```diff
@@ -0,0 +1,104 @@
+use linkerd_identity as id;
+use std::convert::TryFrom;
+use std::time::SystemTime;
+use std::{io, sync::Arc};
+use tokio_rustls::rustls::{
+    self,
+    client::{self, ServerCertVerified, ServerCertVerifier},
+    server::ParsedCertificate,
+    Certificate, RootCertStore, ServerName,
+};
+use tracing::trace;
+
+pub(crate) struct AnySanVerifier(Arc<RootCertStore>);
+
+impl AnySanVerifier {
+    pub(crate) fn new(roots: impl Into<Arc<RootCertStore>>) -> Self {
+        Self(roots.into())
+    }
+}
+
+// Note that this logic is almost identical to the implementation of
+// `rustls::client::WebPkiVerifier`` that cab be found here:
+// https://github.com/rustls/rustls/blob/ccb79947a4811412ee7dcddcd0f51ea56bccf101/rustls/src/webpki/server_verifier.rs#L239
+// The only difference is that we omit the step that performs
+// DNS SAN validation. The reason for that stems from the fact that
+// SAN validation in rustls is limited to DNS SANs only while we
+// want to support alternative SAN types (e.g. URI).
```

Comment:
> This is licensed under Apache. Should we mention something?

My understanding of Apache (and MIT, and ISC, any of which rustls can be used under) is that we still have a copyright requirement for derived work. In fact, if we look at the most liberal (and shortest) license, we see that it is indeed the ONLY requirement:

> ISC License (ISC)
> Copyright (c) 2016, Joseph Birr-Pixton &lt;jpixton@gmail.com&gt;
>
> Permission to use, copy, modify, and/or distribute this software for
> any purpose with or without fee is hereby granted, **provided that the
> above copyright notice and this permission notice appear in all copies.**
> 
> THE SOFTWARE IS PROVIDED "AS IS" AND ...

And regardless of the legal requirements, It's best to bias towards crediting others work.

```suggestion
// This is derived from `rustls::client::WebPkiVerifier`.
//
//   Copyright (c) 2016, Joseph Birr-Pixton <jpixton@gmail.com>
// https://github.com/rustls/rustls/blob/ccb79947a4811412ee7dcddcd0f51ea56bccf101/rustls/src/webpki/server_verifier.rs#L239
//
// The only difference is that we omit the step that performs
// DNS SAN validation. The reason for that stems from the fact that
// SAN validation in rustls is limited to DNS SANs only while we
// want to support alternative SAN types (e.g. URI).
```